src/main/java/bio/terra/cli/serialization/userfacing/resource/UFBqDataTable.java

```diff
@@ -0,0 +1,108 @@
+package bio.terra.cli.serialization.userfacing.resource;
+
+import bio.terra.cli.businessobject.resource.BqDataTable;
+import bio.terra.cli.serialization.userfacing.UFResource;
+import bio.terra.cli.service.GoogleBigQuery;
+import bio.terra.cli.utils.UserIO;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import com.google.api.services.bigquery.model.Table;
+import java.io.PrintStream;
+import java.math.BigInteger;
+import java.util.Optional;
+import javax.annotation.Nullable;
+
+/**
+ * External representation of a workspace BigQuery data table resource for command input/output.
+ *
+ * <p>This is a POJO class intended for serialization. This JSON format is user-facing.
+ *
+ * <p>See the {@link bio.terra.cli.businessobject.resource.BqDataTable} class for a dataset's
+ * internal representation.
+ */
+@JsonDeserialize(builder = UFBqDataTable.Builder.class)
+public class UFBqDataTable extends UFResource {
+  public final String projectId;
+  public final String datasetId;
+  public final String dataTableId;
+  public @Nullable final String tableDescription;
+  public final BigInteger numRows;
+
+  /** Serialize an instance of the internal class to the command format. */
+  public UFBqDataTable(BqDataTable internalObj) {
+    super(internalObj);
+    this.projectId = internalObj.getProjectId();
+    this.datasetId = internalObj.getDatasetId();
+    this.dataTableId = internalObj.getDataTableId();
+
+    GoogleBigQuery bigQuery = GoogleBigQuery.fromContextForPetSa();
+    Optional<Table> dataTableOptional = bigQuery.getDataTable(projectId, datasetId, dataTableId);
+    tableDescription = dataTableOptional.map(Table::getDescription).orElse(null);
+    numRows = dataTableOptional.map(Table::getNumRows).orElse(BigInteger.ZERO);
```

Comment:
I think we should distinguish between the case where there are zero rows and the case where we were unsuccessful in looking up the `Table` object. This could happen in the case where one user adds a data table reference, but another user does not have access. (see `unit.BqDatasetNumTables.numTablesForReferencedWithNoAccess` unit test for an example). Additionally, I think it would be good to add similar tests for the `numRows` property of a BQ data table.